### ISSUE TITLE
Weird things happen when sharing a kid

### ISSUE THREAD

**Issue by jkhartshorne** [jkhartshorne/KidTalkIssues#32]
1. I (Josh) shared a child (BinBin) with another account, which I named "Princess".
2. I logged in as Princess, named myself "Helen", and chose an avatar. 
3. I made a recording, tagging both myself and BinBin.
4. I commented on that recording.
5. I logged in as Josh and looked at my scrapbook. I see only "BinBin" in the taggerd avatar list.
6. I see a comment by Helen, using Helen's avatar. 

**What I expect** 
1. For "Princess" to show up in both the tagged avatars and the comments. 
2. Honestly, I don't know what avatar to expect to see, since I never chose one for "Princess" (I actually can't choose one!)

I'm pretty sure we discussed a plan for how to connect the "tagged adults" and "shared accounts". I can imagine several ways of doing this. But it needs to at least be consistent and sensible! I think using the name and avatar each user chose for themselves makes the most sense. 

![Screen Shot 2020-08-09 at 9 53 51 AM](https://user-images.githubusercontent.com/6118115/89734033-7a615400-da27-11ea-8f2d-a6193839a781.png)

**Comment by alexchippendale** [comment]
@jkhartshorne We are looking into this, thank you for the detail.

**Comment by alexchippendale** [comment]
@jkhartshorne I want to clear up this issue and make sure we understand it correctly.

For (1) under "What I expect" it says you expect "Princess" to show up whereas what happens right now is that "Helen" shows up. The reason for this is because "Helen" is the name the invited other adult chose on their account, and the avatar they had when they commented. 

It seems that this issue is essentially the same two adult problem expressed in https://github.com/alexchippendale/KidTalkRemainingTasks/issues/15 . However, in that the expectation seems to say the opposite, "If possible, it should just automatically use the name and avatar that the other adult chooses for themselves." This should mean that "Helen" and her avatar are used, which is what happens now.

I agree that the two adult issue is confusing, so it seems like the things to do here is for both comments and tags are:
(1) to default to what the other user selected as their name and avatar (if they've selected them), and 
(2) make sure when they tag themselves and you log back into your account, you see their tag of themselves with their chosen name and avatar, not just the child's tag that you both share
(3) Finally we should "automatically make anyone you are sharing with be someone you can tag" as suggested in https://github.com/alexchippendale/KidTalkRemainingTasks/issues/15

In my mind that should solve the two adult issue and relieve the confusion. Let me know what you think, thanks!

**Comment by jkhartshorne** [comment]
@alexchippendale - I guess saying what I "expect" is confusing here. I mean that as a naive user, I expect that if I've labeled someone as "Princess" and given them a particular avatar, then they should show up in the scrapbook as "Princess" and with that avatar. 

The problem goes away if I don't choose a name or avatar for them. Under that scenario, when I share with someone, I don't choose a name or avatar for them. I can tag them in recordings. If they've set up a name and avatar, that's what'll be used when I tag them. If they haven't set up a name or avatar, the name will be based off the email address (the first few letters?) and the avatar will be the generic one -- similar to what we do for someone who doesn't have an avatar yet. 

As I wrote in [](https://github.com/alexchippendale/KidTalkRemainingTasks/issues/15), I could go with either solution. In either case, the behavior should match what a naive user might expect. What I was complaining about was the fact that the user currently has no way of predicting what actually happens.

**Comment by alexchippendale** [comment]
@jkhartshorne Understood, that makes sense

**Comment by alexchippendale** [comment]
This is now deployed to staging. 

What happens now is that whenever you invite a user, it will automatically create a tagged adult of the same name. If the email address you invited is already a user, the tagged adult will default to the name and avatar the invited user has selected. If the invited user later changes or selects a name or avatar, the tagged adult's name and avatar will change in sync. 

Basically, we default to whatever the invited user chooses. You can also just create a tagged adult on its own. In that case it is unconnected to an invited user, and just meant for tagging purposes. That user tag will have whatever name and avatar you give it. But if an invitation is linked to a tag, the tag will change in sync to whatever the invited user chooses.
